httpclient5/src/test/java/org/apache/hc/client5/http/psl/TestPublicSuffixMatcher.java

```diff
@@ -55,10 +61,10 @@ void setUp() throws Exception {
     @Test
     void testGetDomainRootAnyType() {
         // Private
-        Assertions.assertEquals("xx", matcher.getDomainRoot("example.XX"));
-        Assertions.assertEquals("xx", matcher.getDomainRoot("www.example.XX"));
-        Assertions.assertEquals("xx", matcher.getDomainRoot("www.blah.blah.example.XX"));
-        Assertions.assertEquals("appspot.com", matcher.getDomainRoot("example.appspot.com"));
+        Assertions.assertEquals("xx", matcher.getDomainRoot("example.XX", true));
+        Assertions.assertEquals("xx", matcher.getDomainRoot("www.example.XX", true));
+        Assertions.assertEquals("xx", matcher.getDomainRoot("www.blah.blah.example.XX", true));
+        Assertions.assertEquals("appspot.com", matcher.getDomainRoot("example.appspot.com", true));
```

Comment:
It'd be good to copy all the legacy mode tests to a separate section (or sections) and (attempt to) explain what behavior it's "supposed" to have.

Then we can change these versions to use non-legacy mode, and assert the correct result.

Reply:
Sure, I'll make a version of each of these methods that does that.

Reply:
I'll push this in a bit, it's a good idea as this shows the difference between legacy and new behaviour and might show up some edge cases we haven't covered...